pkg/cachemanager/cachemanager_test.go

```diff
@@ -0,0 +1,373 @@
+	require.False(t, cfClient.Contains(expected))
+
+	// create a second GVK
+	podGVK := schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Pod",
+	}
+	// Create pods to test for
+	pod := unstructuredFor(podGVK, "pod-1")
+	require.NoError(t, c.Create(ctx, pod), "creating Pod pod-1")
+
+	pod2 := unstructuredFor(podGVK, "pod-2")
+	require.NoError(t, c.Create(ctx, pod2), "creating Pod pod-2")
+
+	pod3 := unstructuredFor(podGVK, "pod-3")
+	require.NoError(t, c.Create(ctx, pod3), "creating Pod pod-3")
+
+	cacheManager.watchedSet.Add(podGVK)
+	require.NoError(t, cacheManager.syncGVK(ctx, configMapGVK))
+	require.NoError(t, cacheManager.syncGVK(ctx, podGVK))
+
+	expected = map[CfDataKey]interface{}{
+		{Gvk: configMapGVK, Key: "default/config-test-1"}: nil,
+		{Gvk: configMapGVK, Key: "default/config-test-2"}: nil,
+		{Gvk: podGVK, Key: "default/pod-1"}:               nil,
+		{Gvk: podGVK, Key: "default/pod-2"}:               nil,
+		{Gvk: podGVK, Key: "default/pod-3"}:               nil,
+	}
+
+	require.Equal(t, 5, cfClient.Len())
+	require.True(t, cfClient.Contains(expected))
+
+	// cleanup
+	require.NoError(t, c.Delete(ctx, cm), "deleting ConfigMap config-test-1")
+	require.NoError(t, c.Delete(ctx, cm2), "deleting ConfigMap config-test-2")
+	require.NoError(t, c.Delete(ctx, pod), "deleting Pod pod-1")
+	require.NoError(t, c.Delete(ctx, pod3), "deleting Pod pod-3")
+	require.NoError(t, c.Delete(ctx, pod2), "deleting Pod pod-2")
+}
+
+// TestCacheManager_wipeCacheIfNeeded_excluderChanges tests that we can remove gvks that were not previously process excluded but are now.
+func TestCacheManager_wipeCacheIfNeeded_excluderChanges(t *testing.T) {
+	cacheManager, ctx := unitCacheManagerForTest(t)
+	cfClient, ok := cacheManager.cfClient.(*FakeCfClient)
+	require.True(t, ok)
+
+	// seed gvks
+	configMapGVK := schema.GroupVersionKind{Group: "", Version: "v1", Kind: "ConfigMap"}
+	cm := unstructuredFor(configMapGVK, "config-test-1")
+	cm.SetNamespace("excluded-ns")
+	_, err := cfClient.AddData(ctx, cm)
+	require.NoError(t, err, "adding ConfigMap config-test-1 in cfClient")
+	cacheManager.watchedSet.Add(configMapGVK)
+
+	podGVK := schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"}
+	pod := unstructuredFor(configMapGVK, "pod-test-1")
+	pod.SetNamespace("excluded-ns")
+	_, err = cfClient.AddData(ctx, pod)
+	require.NoError(t, err, "adding Pod pod-test-1 in cfClient")
+	cacheManager.watchedSet.Add(podGVK)
+
+	cacheManager.ExcludeProcesses(newSyncExcluderFor("excluded-ns"))
+	cacheManager.wipeCacheIfNeeded(ctx)
+
+	// the cache manager should not be watching any of the gvks that are now excluded
+	require.False(t, cfClient.HasGVK(configMapGVK))
+	require.False(t, cfClient.HasGVK(podGVK))
+	require.False(t, cacheManager.excluderChanged)
+}
+
+// TestCacheManager_AddObject_RemoveObject tests that we can add/ remove objects in the cache.
+func TestCacheManager_AddObject_RemoveObject(t *testing.T) {
```

Comment:
This test and the one below give us limited testing of `AddObject()` and couple it to `RemoveObject`.  Notably, we leave out parts of what `AddObject` does:

1. AddObject won't add a non-watched GVK's object
2. AddObject updates the syncMetricsCache, but only if the object is a watched GVK
3. AddObject updates the tracker, regardless if the GVK is watched or not

This is another example of the value of table-driven tests.  Such tests force us to define a way to compare the before and after states, thus providing an easy way to define and execute the various permutations of our logic via manipulations of the before state and the function's parameters.

Reply:
broke up the tests into tts and decoupled. regarding the tracker checking, i'd recommend we either add them in higher level tests or as a follow up after we add the `syncset_controller`